### DISCUSSION TITLE
Created / Deleted / Updated record Realtime API?

### DISCUSSION THREAD

**Discussion by calumk** [pocketbase/pocketbase#2685] [Q&A] [answered]
Am i missing something obvious - how do i tell if a record had been created  or updated /or deleted using the subscribe realtime api? 
The data looks the same?

Shouldnt there be an "action" parameter or something? 

EG : Top is created, then the next is deleted? but how can you tell? 
```

{
  collectionId: 'ptdb9hvi90ps0kj',
  collectionName: 'things',
  created: '2023-06-11 14:19:19.735Z',
  field: 'second ',
  field1: 'record',
  id: 'qlyumnyrm4prz80',
  updated: '2023-06-11 14:19:19.735Z'
}
{
  collectionId: 'ptdb9hvi90ps0kj',
  collectionName: 'things',
  created: '2023-06-11 14:19:19.735Z',
  field: 'second ',
  field1: 'record',
  id: 'qlyumnyrm4prz80',
  updated: '2023-06-11 14:19:19.735Z'
}


```

**Comment by ganigeorgiev** [comment, marked as answer]
The subscribe event message should have `action` and `record` fields:

```js
await pb.collection("example").subscribe("*", (e) {
    console.log(e.action); // create/update/delete
    console.log(e.record);
})
```

_Note that the `delete` event could fire before the actual record deletion from the database. This is fixed in the `develop` branch._

**Comment by calumk** [comment]
Thanks as always @ganigeorgiev  

This doesnt appear to be documented anywhere  - its not mentioned on : 
https://pocketbase.io/docs/api-realtime
https://github.com/pocketbase/js-sdk

i guess i would have found it if i logged `e` instead of `e.record` , but might be worth a quick addition to the docs.

**Reply by ganigeorgiev** [reply]
I'll add `console.log(e.action)` to the docs and examples with the next release (_the site docs will have a major refactoring in relation to the added js app hooks support_).
